src/Raven.Server/Documents/Sharding/Handlers/Processors/Subscriptions/ShardedSubscriptionsHandlerProcessorForTrySubscription.cs

```diff
@@ -10,14 +10,15 @@
 
 namespace Raven.Server.Documents.Sharding.Handlers.Processors.Subscriptions
 {
-    internal class ShardedSubscriptionsHandlerProcessorForTrySubscription : AbstractSubscriptionsHandlerProcessorForTrySubscription<ShardedDatabaseRequestHandler, TransactionOperationContext>
+    internal class ShardedSubscriptionsHandlerProcessorForTrySubscription : AbstractSubscriptionsHandlerProcessorForTrySubscription<ShardedSubscriptionsHandler, TransactionOperationContext>
     {
-        public ShardedSubscriptionsHandlerProcessorForTrySubscription([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler)
+        public ShardedSubscriptionsHandlerProcessorForTrySubscription([NotNull] ShardedSubscriptionsHandler requestHandler) : base(requestHandler)
         {
         }
 
         protected override async ValueTask TryoutSubscriptionAsync(TransactionOperationContext context, SubscriptionConnection.ParsedSubscription subscription, SubscriptionTryout tryout, int pageSize)
         {
+            RequestHandler.ThrowOnRevisionsIfNeeded(subscription.Revisions);
```

Comment:
In `AbstractSubscriptionsHandlerProcessorForTrySubscription` create a method `protected virtual AssertSubscription(ParsedSubscription subscription)` and override it here. Much cleaner approach.

Reply:
then I will have to create same abstract method in `AbstractSubscriptionsHandlerProcessorForPostSubscription` &  `AbstractSubscriptionsHandlerProcessorForPutSubscription` 
putting it `ShardedSubscriptionsHandler` let me use the same code in all the 3 processors (try, post, put)

Reply:
Yes, but inside that abstract method you can use static method from the sharded processor

Reply:
created new helper class